src/imageResource.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+require('openseadragon');
+
+var ImageResource = function(config, parent, dispatcher) {
+  this.needed = config.needed || false;
+  this.visible = config.visible || false;
+  this.clipRegion = config.clipRegion;
+  this.opacity = config.opacity || 1;
+  this.x = config.x || 0;
+  this.y = config.y || 0;
+  this.height = config.height || 1;
+  this.width = config.width || 1;
+  this.zIndex = config.zIndex || 0;
+  this.tileSource = config.tileSource;
+  this.dynamic = config.dynamic || false;
+  this.imageType = config.imageType || "main"; // can be 'main', 'alternate', or 'detail'
+  this.status = 'initialized'; // can be 'requested', 'received', 'pending','shown', or 'failed'
+  this.parent = parent;
+  this.dispatcher = dispatcher;
+};
+
+ImageResource.prototype = {
+  hide: function(immediately) {
+    this.previousOpacity = this.opacity;
```

Comment:
Probably a better way to handle this would be to have an `_updateOpacity` method that keys off of both the `opacity` and `visible` properties. It would be called from hide, show, and setOpacity. That way you don't have to mess with previousOpacity and worry about it getting stale.